.github/workflows/upgrade.yml

```diff
@@ -0,0 +1,59 @@
+# SPDX-FileCopyrightText: 2025 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: On-Prem Upgrade
+
+# Only run at most 1 workflow concurrently per PR, unlimited for branches
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.event.pull_request.number || github.sha }}
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
+
+# Manual Trigger
+on:
+  workflow_dispatch:
+
+jobs:
+  upgrade-from-version:
+    name: Upgrade from version
+    strategy:
+      matrix:
+        # Add comma separated Git tags here to test upgrade from different versions, e.g. [2.0.0, v3.0.0]
+        version-tags: [v3.0.0]
```

Comment:
Maybe these can be provided as input? e.g., 

inputs:

Reply:
Wasn't aware. We have more changes to follow, so I'll look into inputs.